pkg/sql/executor_ir.go

```diff
@@ -0,0 +1,424 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	pb "sqlflow.org/sqlflow/pkg/server/proto"
+	"sqlflow.org/sqlflow/pkg/sql/codegen"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/tensorflow"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
+)
+
+// EndOfExecution will push to the pipe when one SQL statement execution is finished.
+type EndOfExecution struct {
+	StartTime int64
+	EndTime   int64
+	Statement string
+}
+
+// RunIR execute a list of parsed SQL statement IRs and merge the results.
+func RunIR(programIR codegen.SQLProgramIR, db *DB, cwd, modelDir string, session *pb.Session) *PipeReader {
```

Comment:
Remove `cwd` parameter. Steps shouldn't be sharing the same directory.

Reply:
May need to remove `cwd` parameter from `generatePredictIR` and `generateAnalyzeIR` so that we can separate generating IRs and run them.

Reply:
Sounds good.